clients/hadoopfs/src/main/java/io/lakefs/storage/LakeFSFileSystemOutputStream.java

```diff
@@ -0,0 +1,73 @@
+package io.lakefs.storage;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.hadoop.fs.Path;
+
+import io.lakefs.LakeFSClient;
+import io.lakefs.LakeFSFileSystem;
+import io.lakefs.clients.api.ApiException;
+import io.lakefs.clients.api.StagingApi;
+import io.lakefs.clients.api.model.StagingLocation;
+import io.lakefs.clients.api.model.StagingMetadata;
+import io.lakefs.utils.ObjectLocation;
+
+public class LakeFSFileSystemOutputStream extends OutputStream {
+    private final HttpURLConnection connection;
+    private final ByteArrayOutputStream buffer;
+    private final StagingLocation stagingLocation;
+    private final ObjectLocation objectLoc;
+    private final AtomicBoolean isClosed = new AtomicBoolean(false);
+    private final LakeFSFileSystem lfs;
+    private final LakeFSClient lakeFSClient;
+
+    public LakeFSFileSystemOutputStream(LakeFSFileSystem lfs, LakeFSClient lfsClient, HttpURLConnection connection, ObjectLocation objectLoc, StagingLocation stagingLocation) throws IOException {
```

Comment:
This needs a big TODO not to keep all data in memory.

Reply:
Done